raiden/storage/sqlite.py

```diff
@@ -477,40 +557,35 @@ def update_state_changes(self, state_changes_data: List[Tuple[str, int]]) -> Non
         )
         self.maybe_commit()
 
-    def get_statechanges_by_identifier(
-        self, from_identifier: StateChangeID, to_identifier: Union[StateChangeID, str]
-    ) -> List[str]:
-        typecheck(from_identifier, T_StateChangeID)
-
-        if not (
-            to_identifier == "latest" or isinstance(to_identifier, T_StateChangeID)
-        ):  # pragma: no unittest
-            raise ValueError("to_identifier must be an integer or 'latest'")
+    def get_statechanges_records_by_range(
+        self, db_range: Range[StateChangeID]
+    ) -> List[StateChangeEncodedRecord]:
+        if not isinstance(db_range, Range):  # pragma: no unittest
+            raise ValueError("db_range must be an Range")
 
         cursor = self.conn.cursor()
 
-        if to_identifier == "latest":
-            cursor.execute(
-                "SELECT data FROM state_changes WHERE identifier >= ? ORDER BY identifier ASC",
-                (from_identifier,),
-            )
-        else:
-            cursor.execute(
-                "SELECT data FROM state_changes WHERE identifier "
-                "BETWEEN ? AND ? ORDER BY identifier ASC",
-                (from_identifier, to_identifier),
-            )
+        query = (
+            "SELECT identifier, data "
+            "FROM state_changes "
+            "WHERE identifier "
+            "BETWEEN ? AND ? "
```

Comment:
does the `BETWEEN` operator work bi-directionally? IOW are the queries:
```sql
[...]
BETWEEN LOW_STATECHANGE_ULID AND HIGH_STATECHANGE_ULID
[...]
```
and
```sql
[...]
BETWEEN HIGH_STATECHANGE_ULID AND LOW_STATECHANGE_ULID
[...]
```
equivalent? Otherwise I suggest a validation for the range above!

Reply:
> does the BETWEEN operator work bi-directionally?

It does not:

> "x BETWEEN y AND z" is equivalent to "x>=y AND x<=z" 

https://www.sqlite.org/lang_expr.html#between

That is why I introduced the type `Range`, it cannot be constructed if the range is invalid:

https://github.com/raiden-network/raiden/pull/4156/files#diff-34fcdfdd66b47942cc02bf2a12fa4925R48